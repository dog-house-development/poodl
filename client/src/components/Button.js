import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    onClick: PropTypes.func.isRequired,
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    type: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
    content: PropTypes.string.isRequired
};

const defaultProps = {
    size: 'medium',
    type: 'primary'
};

class Button extends Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(e) {
        e.preventDefault();
        this.props.onClick();
    }

    render() {
        return (
            <button onClick={this.handleClick} className={`button ${this.props.size} ${this.props.type}`}>
                {this.props.content}
            </button>
        );
    }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
