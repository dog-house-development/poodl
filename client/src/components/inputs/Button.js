import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    onClick: PropTypes.func,
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    kind: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
    content: PropTypes.string.isRequired,
    type: PropTypes.string
};

const defaultProps = {
    size: 'medium',
    kind: 'primary'
};

class Button extends Component {
    // constructor(props) {
    //     super(props);
    // }

    render() {
        return (
            <button
                onClick={this.props.onClick}
                className={`button ${this.props.size} ${this.props.kind}`}
                type={this.props.type} >
                {this.props.content}
            </button>
        );
    }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
