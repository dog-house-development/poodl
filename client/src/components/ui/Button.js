import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
    onClick: PropTypes.func,
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    kind: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
    content: PropTypes.string.isRequired,
    type: PropTypes.string,
    formButton: PropTypes.bool,
    dropdownButton: PropTypes.bool
};

const defaultProps = {
    size: 'medium',
    kind: 'primary',
    dropdownButton: false
};

class Button extends Component {
    render() {
        return (
            <button
                onClick={this.props.onClick}
                className={`button ${this.props.size} ${this.props.kind} ${this.props.formButton ? 'form-button' : ''}`}
                type={this.props.type}
            >
                {this.props.content}
            </button>
        );
    }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
