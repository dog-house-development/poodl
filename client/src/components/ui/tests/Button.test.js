import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import Button from '../Button';

configure({ adapter: new Adapter() });

describe('Button tests', () => {
    let wrapper, instance;
    const setInstanceAndWrapper = (_props = {}) => {
        const props = _.assign(
            {},
            {
                onClick: () => {},
                size: 'medium',
                kind: 'primary',
                content: 'Press me',
                type: 'normal'
            },
            _props
        );
        wrapper = shallow(<Button {..._.assign({}, props)} />);
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('getIconMarkup', () => {
        it('should render loading', () => {
            setInstanceAndWrapper({ loading: true });
            expect(wrapper).toMatchSnapshot();
        });
        it('should run getIconMarkup correctly', () => {
            setInstanceAndWrapper({ icon: 'account_box' });
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('getContent', () => {
        it('should render getContent correctly', () => {
            setInstanceAndWrapper();
            expect(wrapper).toMatchSnapshot();
        });
        it('should render loading', () => {
            setInstanceAndWrapper({ loading: true });
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('render', () => {
        it('should render normal button correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('should render form-button correctly', () => {
            setInstanceAndWrapper({ formButton: true });
            expect(wrapper).toMatchSnapshot();
        });
    });
});
