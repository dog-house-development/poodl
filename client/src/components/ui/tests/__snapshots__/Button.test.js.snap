// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests getContent should render getContent correctly 1`] = `
<button
  className="button medium primary"
  onClick={[Function]}
  type="normal"
>
   
  Press me
</button>
`;

exports[`Button tests getContent should render loading 1`] = `
<button
  className="button medium primary"
  loading={true}
  onClick={[Function]}
  type="normal"
>
  <Loading
    content="Loading..."
    kind="primary"
    size="medium"
  />
</button>
`;

exports[`Button tests getIconMarkup should render loading 1`] = `
<button
  className="button medium primary"
  loading={true}
  onClick={[Function]}
  type="normal"
>
  <Loading
    content="Loading..."
    kind="primary"
    size="medium"
  />
</button>
`;

exports[`Button tests getIconMarkup should run getIconMarkup correctly 1`] = `
<button
  className="button medium primary icon"
  onClick={[Function]}
  type="normal"
>
  <i
    className="material-icons button-icon"
  >
    account_box
  </i>
   
  Press me
</button>
`;

exports[`Button tests render should render form-button correctly 1`] = `
<button
  className="button medium primary form-button"
  onClick={[Function]}
  type="normal"
>
   
  Press me
</button>
`;

exports[`Button tests render should render normal button correctly 1`] = `
<button
  className="button medium primary"
  onClick={[Function]}
  type="normal"
>
   
  Press me
</button>
`;
